src/components/IGVBrowser.js

```diff
@@ -1,59 +1,60 @@
 import _ from 'lodash/fp'
-import { Component } from 'src/libs/wrapped-components'
-import { createRef } from 'react'
-import colors from 'src/libs/colors'
-import { Ajax } from 'src/libs/ajax'
+import { useRef, useState } from 'react'
 import { div } from 'react-hyperscript-helpers'
-import igv from 'igv/dist/igv.js'
+import { centeredSpinner } from 'src/components/icons'
+import { Ajax } from 'src/libs/ajax'
+import colors from 'src/libs/colors'
+import { reportError } from 'src/libs/error'
+import * as Utils from 'src/libs/utils'
+
 
 const igvStyle = {
-  paddingTop: '10px',
-  paddingBottom: '10px',
-  margin: '8px',
+  padding: '10px 0',
+  margin: 8,
   border: `1px solid ${colors.gray[5]}`
 }
 
-export class IGVBrowser extends Component {
-  constructor(props) {
-    super(props)
-    this.containerRef = createRef()
-  }
-
-  getTrackType(filePath) {
-    const fileTypeToTrack = {
-      'bam': 'alignment',
-      'cram': 'alignment',
-      'bed': 'annotation',
-      'vcf': 'variant'
+export const IGVBrowser = ({ selectedFiles, refGenome }) => {
+  const containerRef = useRef()
+  const [loadingIgv, setLoadingIgv] = useState(true)
+
+  Utils.useOnMount(async () => {
+    const getTrack = filePath => {
+      const fileTypeToTrack = {
+        'bam': 'alignment',
+        'cram': 'alignment',
+        'bed': 'annotation',
+        'vcf': 'variant'
+      }
+
+      return {
+        type: fileTypeToTrack[_.last(filePath.split('.'))],
+        name: filePath,
+        url: filePath,
+        // igv.js will automatically find the index file, but not for crams
+        indexURL: filePath.endsWith('cram') ? filePath + '.crai' : ''
+      }
     }
-    return fileTypeToTrack[_.last(filePath.split('.'))]
-  }
-
-  getTrack(filePath) {
-    return {
-      type: this.getTrackType(filePath),
-      name: filePath,
-      url: filePath,
-      // igv.js will automatically find the index file, but not for crams
-      indexURL: filePath.endsWith('cram') ? filePath + '.crai' : ''
+
+    const options = {
+      genome: refGenome,
+      tracks: _.map(getTrack, selectedFiles)
     }
-  }
-
-  componentDidMount() {
-    const { selectedFiles, refGenome } = this.props
-    const tracks = _.map(v => this.getTrack(v))(selectedFiles)
-    const myOptions =
-      {
-        genome: refGenome,
-        tracks: tracks
-      }
-    igv.setGoogleOauthToken(Ajax().User.token())
```

Comment:
Apparently this way of setting the token is deprecated, so I changed to the new syntax.

Reply:
Interesting, I'll move to update https://github.com/igvteam/igv.js/wiki/OAuth-Support then, which is where I saw this.

Reply:
Figures. The deprecation is [in the doc comment](https://github.com/igvteam/igv.js/blob/master/js/oauth.js#L83-L90), but the old way is still in use [here](https://github.com/igvteam/igv.js/blob/master/js/igvxhr.js), for example.